src/common/services/applemusic/resolve.ts

```diff
@@ -1,88 +1,51 @@
 import { asArray } from '../../utils/array'
-import { stringToDate } from '../../utils/datetime'
+import { secondsToString, stringToDate } from '../../utils/datetime'
 import { fetch } from '../../utils/fetch'
-import { decode } from '../../utils/io-ts'
 import { getReleaseType } from '../../utils/music'
-import { isDefined, isUndefined } from '../../utils/types'
-import { ResolveFunction } from '../types'
-import { ReleaseHolder } from './codec'
+import { ResolveData, ResolveFunction } from '../types'
+import { Release, ReleaseHolder } from './codec'
 
-const getUrl = (document_: Document) =>
-  document_.querySelector<HTMLLinkElement>('link[rel="canonical"]')?.href
-
-const getArtists = (document_: Document) =>
-  [...document_.querySelectorAll('.product-creator a')]
-    .map((element) => element.textContent?.trim())
-    .filter(isDefined)
-
-const getDate = (document_: Document) => {
-  const dateString = document_.querySelector<HTMLMetaElement>(
-    'meta[property="music:release_date"]'
-  )?.content
-  return isUndefined(dateString) ? undefined : stringToDate(dateString)
-}
-
-const getTracks = (document_: Document) =>
-  [...document_.querySelectorAll('.songs-list-row')].map((element) => {
-    const position = element
-      .querySelector('.songs-list-row__column-data')
-      ?.textContent?.trim()
-    const title = element
-      .querySelector('.songs-list-row__song-name')
-      ?.textContent?.trim()
-    const duration = element
-      .querySelector('.songs-list-row__length')
-      ?.textContent?.trim()
-    return { position, title, duration }
-  })
-
-const getCoverArt = (document_: Document) => {
-  const element = document_.querySelector<HTMLSourceElement>(
-    '.product-lockup picture source[type="image/jpeg"]'
-  )
-  const firstLink = element?.srcset.split(', ')[0]?.split(' ')[0]
-  if (isUndefined(firstLink)) return undefined
-
-  // replace the last occurrence of 000x000 with 2400x2400
-  const splits = firstLink.split('/')
-  splits[splits.length - 1] =
-    splits[splits.length - 1]?.replace(/\d+x\d+/, '2400x2400') ?? ''
-  return splits.join('/')
+const getArtists = (release: Release) => {
+  if (release.relationships.artists.data.length > 0) {
+    return release.relationships.artists.data.map((artist) => {
+      artist.attributes.name
```

Comment:
remove brackets around `artist.attributes.name` (it's currently not returning anything). this should also remove the need to cast the return of the `resolve` function to `ResolveData` on line 69